### DISCUSSION TITLE
One Click into Codespaces - Codespaces check-in

### DISCUSSION THREAD

**Discussion by otter-computer** [community/community#53794] [Codespaces]
Hey everyone, it’s time for the latest Codespaces check-in! This week we’ve got some brand new Codespaces features that should make managing, using, and sharing your Codespaces easier! Plus, a note about some recent changes to the community.
### [Recommended secrets](https://github.blog/changelog/2023-04-24-one-click-into-github-codespaces/#recommended-secrets)
Dev Container Secrets allow project maintainers to define all the necessary secrets a contributor must add for a one-click development experience. Once your secrets are defined in your project’s `devcontainer.json` file, contributors will be asked to add these required secrets on Codespace creation letting them get into the code ASAP!
### [Codespace share link](https://github.blog/changelog/2023-04-24-one-click-into-github-codespaces/#creating-a-share-link--resuming-a-codespace)
You can now create a direct link to Codespace creation for your project. This link allows direct, one-click creation of a Codespace for a contributor — even from outside of GitHub! Create these links from inside the three-dot menu of the Codespace creation menu inside your repository.
### [Resuming a Codespace](https://github.blog/changelog/2023-04-24-one-click-into-github-codespaces/#creating-a-share-link--resuming-a-codespace)
Codespace deep-links will allow you to resume an existing Codespace — if you have any — instead of forcing you to create a new one. This should make it much easier for you to get back to your code that’s already in-flight.
### [Codespace hotkey](https://github.blog/changelog/2023-04-24-one-click-into-github-codespaces/#jumping-into-development-from-a-repository-with-a-comma)
Finally, you can now quick-resume a Codespace from your repository with a keyboard shortcut. Simply tap `,` to jump to the Codespace resume or creation flow and get back to coding fast 🌩
## 🆕 Community Interest Categories 🧠
Take your skills that much further without even leaving the community, we added the [Programming Help](https://github.com/orgs/community/discussions/categories/programming-help?discussions_q=is%3Aopen+sort%3Adate_created+category%3A%22Programming+Help%22) and [Machine Learning](https://github.com/orgs/community/discussions/categories/machine-learning-and-mlops?discussions_q=is%3Aopen+sort%3Adate_created+category%3A%22Machine+Learning+and+MLOps%22) categories to discuss your projects, programming, AI, and more!

**Comment by Nabeelperson** [comment]
The keyboard shortcut is extremely annoying, please allow us to configure not having the period open a window that involves several redirects and makes it hard to get back to what you were actually doing. 
@otter-computer
